integration-tests/src/test/java/oracle/weblogic/kubernetes/ItRetryImprovements.java

```diff
@@ -313,26 +337,100 @@ void testRetryOccursAsExpectedAndThrowWarning() {
+        .append(".*to\\s*force\\s*another\\s*retry.*").toString();
+
+    // verify that the operator stops retry after failure retry limit minutes expired
+    testUntil(() -> findStringInDomainStatusMessage(domainNamespace, domainUid, retryDoneMsgRegex),
+        logger, "The operator stops retry after failure retry limit minutes expired");
+  }
+
+  /**
+   * Create a model-in-image domain with bad model file from configmap
+   * the domain should fail to start with SEVERE error in introspector log and Operator log
+   * and the Operator should start retrying. Verify that error in introspector log is logged to Operator log.
+   */
+  @Test
+  @DisplayName("Creating a domain with bad model file from configmap and "
+      + "verify that retry occurs and error in introspector logged in the Operator log")
+  void testRetryOccursAndErrorFromIntrospectorLoggedInOperator() throws Exception {
+    Long failureRetryLimitMinutes = Long.valueOf("1");
+    int replicaCount = 2;
+
+    String badModelFileCm = "bad-model-in-cm";
+    String badModelFileName = "bad-model-file.yaml";
+    Path badModelFile = Paths.get(MODEL_DIR, badModelFileName);
+
+    logger.info("Creating a domain resource with bad model file from configmap");
+    DomainResource domain =
+        createDomainResourceForRetryTestWithConfigMap(failureRetryLimitMinutes,
+            replicaCount, badModelFile, badModelFileCm);
+    createDomainAndVerify(domain, domainNamespace);
+
+    String createDomainFailedMsgRegex = new StringBuffer(".*SEVERE.*createDomain\\s*was\\s*unable\\s*to\\s*load.*")
+        .append(badModelFileName).toString();
+    String retryDoneMsgRegex = new StringBuffer(".*operator\\s*failed\\s*after\\s*retrying\\s*for\\s*")
+        .append(failureRetryLimitMinutes.toString())
+        .append("\\s*minutes.*Please\\s*resolve.*update\\s*domain.spec.introspectVersion\\s*")
+        .append(".*to\\s*force\\s*another\\s*retry.*").toString();
+
+    // verify that retryDoneMsgRegex message found in domain status message
+    testUntil(() -> findStringInDomainStatusMessage(domainNamespace, domainUid, retryDoneMsgRegex),
+        logger, "{0} is found in domain status message", retryDoneMsgRegex);
+
+    // verify that SEVERE and createDomainFailedMsgRegex message found in Operator log
+    testUntil(() -> checkPodLogContainsRegex(createDomainFailedMsgRegex, operatorPodName, opNamespace),
+        logger, "{0} is found in Operator log", createDomainFailedMsgRegex);
+
+    // verify that SEVERE and createDomainFailedMsgRegex message found in Operator log
+    testUntil(() -> checkInUncompletedIntroPodLogContainsRegex(createDomainFailedMsgRegex,
+        domainUid, domainNamespace),
+        logger, "{0} is found in introspector log", createDomainFailedMsgRegex);
+
+    Callable<Boolean> configMapExist = assertDoesNotThrow(() -> configMapExist(domainNamespace, badModelFileCm));
+
+    if (configMapExist.call().booleanValue()) {
```

Comment:
You can use 
if (configMapExist(domainNamespace, badModelFileCm)) directly.

Reply:
Sorry I can't. configMapExist.call().booleanValue() returns Callable<Boolean> but if requires boolean